met-api/src/met_api/services/comment_service.py

```diff
@@ -0,0 +1,75 @@
+
+"""Service for comment management."""
+from met_api.constants.comment_status import Status
+from met_api.models.comment import Comment
+from met_api.schemas.comment import CommentSchema
+from met_api.schemas.submission import SubmissionSchema
+from met_api.schemas.survey import SurveySchema
+
+
+class CommentService:
+    """Comment management service."""
+
+    otherdateformat = '%Y-%m-%d'
+
+    @staticmethod
+    def get_comment(comment_id, user_id) -> CommentSchema:
+        """Get Comment by the id."""
+        comment = Comment.get_comment(comment_id)
+
+        if comment:
+            if user_id is None and comment.get('status_id', None) != Status.Accepted:
+                # Non authenticated users only have access to accepted comments
+                return None
+
+        return comment
+
+    @classmethod
+    def get_comments_by_survey_id(cls, user_id, survey_id):
+        """Get all comments."""
+        filters = {}
+        if not user_id:
+            filters['status_id'] = Status.Accepted
+        return Comment.get_comments_by_survey_id_query(survey_id, **filters)
+
+    @classmethod
+    def create_comments(cls, comments: list):
+        """Create comment."""
+        for comment in comments:
+            cls.validate_fields(comment)
+
+        return Comment.bulk_create_comment(comments)
+
+    @staticmethod
+    def validate_fields(data):
+        """Validate all fields."""
+        # Will empty text return False
+        empty_fields = [not data[field] for field in ['text', 'survey_id']]
+
+        if any(empty_fields):
+            raise ValueError('Some required fields for comments are missing')
+
+    @staticmethod
+    def extract_comments(survey_submission: SubmissionSchema, survey: SurveySchema):
+        """Extract comments from survey submission."""
+        survey_form = survey.get('form_json', {})
+        components = list(survey_form.get('components', []))
+        if len(components) == 0:
+            return []
+
+        comments_keys = [
+            component.get(
+                'key',
+                None) for component in components if component.get(
+                'inputType',
+                None) == 'text']
```

Comment:
looks good, does text apply to any string?

Reply:
we have control over this. Basically any form.io component that we have if it is an input component there is an inputType field that we can define for it. For now textfield and  textarea have inputType text. If we wanted to add another form.io component we have to decide if we treat it as comment (and give it input type text) or something else and choose another name for it